packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -192,25 +295,74 @@ class TextMapPropagator {
   }
 
   _extractTraceparentContext (carrier) {
-    if (!this._config.experimental.traceparent) return null
-
     const headerValue = carrier[traceparentKey]
     if (!headerValue) {
       return null
     }
-    const matches = headerValue.match(traceparentExpr)
+    const matches = headerValue.trim().match(traceparentExpr)
     if (matches.length) {
-      return new DatadogSpanContext({
-        traceId: id(matches[2], 16),
-        spanId: id(matches[3], 16),
-        sampling: { priority: matches[4] === '01' ? 1 : 0 }
+      const [ version, traceId, spanId, flags, tail ] = matches.slice(1)
+      const tracestate = TraceState.fromString(carrier.tracestate)
+      if (invalidSegment.test(traceId)) return null
+      if (invalidSegment.test(spanId)) return null
+
+      // Version ff is considered invalid
+      if (version === 'ff') return null
+
+      // Version 00 should have no tail, but future versions may
+      if (tail && version === '00') return null
+
+      const spanContext = new DatadogSpanContext({
+        traceId: id(traceId, 16),
+        spanId: id(spanId, 16),
+        sampling: { priority: flags === '01' ? 1 : 0 },
```

Comment:
Is there a spec for this `flags` entry? Like is it hex representing a bit mask or something?

Reply:
https://www.w3.org/TR/trace-context/#trace-flags

Reply:
Ah, so, shouldn't we perform bitwise logic to ensure the lowest bit is set?

Reply:
Could do, but the current spec only allows that one bit to be set for the sampled flag so `00` and `01` are the only possible values at the moment. Maybe best to change just so we don't have to change it in the future though. 🤔